cargo-shuttle/src/args.rs

```diff
@@ -1,116 +1,122 @@
 }
 
-#[derive(StructOpt)]
+#[derive(Parser)]
 pub enum Command {
-    #[structopt(about = "deploy a shuttle project")]
+    #[clap(help = "deploy a shuttle project")]
     Deploy(DeployArgs),
-    #[structopt(about = "create a new shuttle project")]
+    #[clap(help = "create a new shuttle project")]
     Init(InitArgs),
-    #[structopt(about = "view the status of a shuttle project")]
+    #[clap(help = "view the status of a shuttle project")]
     Status,
-    #[structopt(about = "view the logs of a shuttle project")]
+    #[clap(help = "view the logs of a shuttle project")]
     Logs,
-    #[structopt(about = "delete the latest deployment for a shuttle project")]
+    #[clap(help = "delete the latest deployment for a shuttle project")]
     Delete,
-    #[structopt(about = "create user credentials for the shuttle platform")]
+    #[clap(help = "create user credentials for the shuttle platform")]
     Auth(AuthArgs),
-    #[structopt(about = "login to the shuttle platform")]
+    #[clap(help = "login to the shuttle platform")]
     Login(LoginArgs),
-    #[structopt(about = "run a shuttle project locally")]
+    #[clap(help = "run a shuttle project locally")]
     Run(RunArgs),
 }
 
-#[derive(StructOpt)]
+#[derive(Parser)]
 pub struct LoginArgs {
-    #[structopt(long, about = "api key for the shuttle platform")]
+    #[clap(long, help = "api key for the shuttle platform")]
     pub api_key: Option<String>,
 }
 
-#[derive(StructOpt)]
+#[derive(Parser)]
 pub struct AuthArgs {
-    #[structopt(about = "the desired username for the shuttle platform")]
+    #[clap(help = "the desired username for the shuttle platform")]
     pub username: String,
 }
 
-#[derive(StructOpt)]
+#[derive(Parser)]
 pub struct DeployArgs {
-    #[structopt(long, about = "allow dirty working directories to be packaged")]
+    #[clap(long, help = "allow dirty working directories to be packaged")]
     pub allow_dirty: bool,
-    #[structopt(long, about = "allows pre-deploy tests to be skipped")]
+    #[clap(long, help = "allows pre-deploy tests to be skipped")]
     pub no_test: bool,
 }
 
-#[derive(StructOpt, Debug)]
+#[derive(Parser, Debug)]
 pub struct RunArgs {
-    #[structopt(long, about = "port to start service on", default_value = "8000")]
+    #[clap(long, help = "port to start service on", default_value = "8000")]
     pub port: u16,
 }
 
-#[derive(StructOpt)]
+#[derive(Parser)]
 pub struct InitArgs {
-    #[structopt(
-        about = "the path to initialize a new shuttle project",
+    #[clap(
+        help = "the path to initialize a new shuttle project",
         parse(try_from_os_str = parse_init_path),
         default_value = ".",
     )]
     pub path: PathBuf,
 }
 
 // Helper function to parse and return the absolute path
-fn parse_path(path: &OsStr) -> Result<PathBuf, OsString> {
-    canonicalize(path).map_err(|e| format!("could not turn {path:?} into a real path: {e}").into())
+fn parse_path(path: &OsStr) -> Result<PathBuf, io::Error> {
```

Comment:
When I just changed the crate from `structopt` to `clap`, I got the following error:

```
error[E0277]: `OsString` doesn't implement `std::fmt::Display`
   --> cargo-shuttle/src/args.rs:39:33
    |
39  |         parse(try_from_os_str = parse_path),
    |                                 ^^^^^^^^^^ `OsString` cannot be formatted with the default formatter
    |
   ::: /Users/xavientois/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/macros.rs:113:39
    |
113 |         let res = $crate::fmt::format($crate::__export::format_args!($($arg)*));
    |                                       ---------------------------------------- in this macro invocation
    |
    = help: the trait `std::fmt::Display` is not implemented for `OsString`
    = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
    = note: this error originates in the macro `$crate::__export::format_args` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `OsString: StdError` is not satisfied
    --> cargo-shuttle/src/args.rs:39:33
     |
39   |         parse(try_from_os_str = parse_path),
     |                                 ^^^^^^^^^^ the trait `StdError` is not implemented for `OsString`
     |
     = note: required because of the requirements on the impl of `From<OsString>` for `Box<(dyn StdError + Sync + std::marker::Send + 'static)>`
     = note: required because of the requirements on the impl of `Into<Box<(dyn StdError + Sync + std::marker::Send + 'static)>>` for `OsString`
note: required by a bound in `Arg::<'help>::validator_os`
    --> /Users/xavientois/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-3.2.6/src/builder/arg.rs:1545:12
     |
1545 |         E: Into<Box<dyn Error + Send + Sync + 'static>>,
     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Arg::<'help>::validator_os`
```

Instead of returning an OsString as the error here, I am wrapping it in an `io::Error` to satisfy this trait bound.